### PR TITLE
llama2 support chat

### PR DESCRIPTION
**Pull request by lylalala**
llama2 model add chat method

### CODE REVIEW

File: tests/models/test_llama2.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        model = Model.from_pretrained(
+            model_dir, device_map='auto', torch_dtype=torch.float16)
+        tokenizer = Llama2Tokenizer.from_pretrained(model_dir)
+
+        inputs = {
+            'utterance': self.utterance,
+            'history': [],
+            'system': self.system
+        }
+        result = model.chat(input=inputs, tokenizer=tokenizer)
```
Comment (tastelikefeet) [inline tests/models/test_llama2.py:30]: Pls add tests for multiple round chat

File: modelscope/models/nlp/llama2/text_generation.py [hunk 2]
```diff
@@ -186,3 +199,42 @@ def _reorder_cache(past_key_values, beam_idx):
...
         return reordered_past
+
+    def chat(self, input: Dict, tokenizer) -> Dict:
+        import copy
+        gen_kwargs = copy.copy(input)
+        if 'utterance' not in input:
+            utterance: str = input['text']
+            gen_kwargs.pop('text')
+        else:
+            utterance: str = input['utterance']
```
Comment (wenmengzhou) [inline modelscope/models/nlp/llama2/text_generation.py:210]: use text instead of utterance

File: tests/models/test_llama2.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        assert isinstance(result['history'], list)
```
Comment (wenmengzhou) [inline tests/models/test_llama2.py:31]: self.assertTrue  for line 31 and 32

File: modelscope/models/nlp/llama2/text_generation.py [hunk 4]
```diff
@@ -186,3 +199,42 @@ def _reorder_cache(past_key_values, beam_idx):
...
+            system=system, utterance=utterance, history=history)
+        inputs = tokenizer(prompt, return_tensors='pt')
+        input_ids = inputs.input_ids.to(self.device)
+        generate_ids = self.generate(input_ids, **gen_kwargs)
+        response = tokenizer.batch_decode(
+            generate_ids,
+            skip_special_tokens=True,
+            clean_up_tokenization_spaces=False)[0]
+        response = response[len(prompt):].strip()
+        history.append((utterance, response))
```
Comment (wenmengzhou) [inline modelscope/models/nlp/llama2/text_generation.py:238]: add history support in pipeline

**Review by wenmengzhou — CHANGES_REQUESTED** [review]

File: tests/models/test_llama2.py (hunk 1 above)
Comment (lylalala) [reply tests/models/test_llama2.py:32]: done

File: modelscope/models/nlp/llama2/text_generation.py (hunk 2 above)
Comment (lylalala) [reply modelscope/models/nlp/llama2/text_generation.py]: done

File: modelscope/models/nlp/llama2/text_generation.py (hunk 4 above)
Comment (lylalala) [reply modelscope/models/nlp/llama2/text_generation.py]: process it in the model for now, and extend it in the pipeline later

**Review by tastelikefeet — APPROVED** [review]

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**